x-pack/plugins/osquery/server/routes/fleet_wrapper/get_agents.ts

```diff
@@ -33,17 +40,81 @@ export const getAgentsRoute = (router: IRouter, osqueryContext: OsqueryAppContex
         },
       },
       async (context, request, response) => {
-        let agents;
+        let esAgents;
+        const query = request.query as ListWithKuery & { showInactive: boolean };
         try {
-          agents = await osqueryContext.service
-            .getAgentService()
-            ?.asInternalUser // @ts-expect-error update types
-            .listAgents(request.query);
+          esAgents = await osqueryContext.service.getAgentService()?.asInternalUser.listAgents({
```

Comment:
I am wondering if this is the interface we should meet here? : https://github.com/elastic/kibana/blob/474c8eaf527aa61b57c05b1ad30b24db75859714/x-pack/plugins/fleet/server/services/agents/crud.ts#L207 
If yes, shouldn't we support more fields?